helpers/verificationHelper.js

```diff
@@ -0,0 +1,45 @@
+const tokenService = require('../services/tokenService');
+const emailService = require('../services/emailService');
+const emailTemplates = require('../services/emailTemplates');
+
+module.exports = {
+  /**
+   * Send verification to a user on sign up
+   * @param {Object} user - user retrieved from the database
+   * @returns {Promise} send email
+   */
+  sendVerificationEmail: function(user) {
+    const token = tokenService.generateToken({ id: user.id }, 60 * 30);
+    const url = process.env.BASEURL + "/api/user/verify/" + token;
+    const mailOptions = emailService.mailOptions(
+        user.email,
+        "Verify your Authors Haven account",
+        emailTemplates.verificationTemplate(url)
+    )
+    // do not send email on test environment
+    if(process.env.NODE_ENV === 'test') return;
+    return emailService.sendMail(mailOptions);
+  },
+  /**
+   * Validate user before verifying or re-sending email
+   * @param {Object} user - user retrieved from the database
+   * @returns {Boolean} false
+   */
+  validateUser: function(user) {
+    if(!user) return { statusCode: 404, data: { status: "error", message: "User not found in the database" } };
+    if(user.verified) return { statusCode: 409, data: { status: "error", message: "Your account has already been verified" }};
+    return false;
+  },
+  /**
+   * Validate email on re-sending verification email
+   * @param {String} email - email passed, on requesting for verification link resend
+   * @returns {Boolean} false
+   */
+  validateEmail: function(email) {
+    if(!email) return { statusCode: 400, data: { status: 'error', message: 'Email not provided' } };
+    if(!/^(([^<>()\[\]\.,;:\s@\"]+(\.[^<>()\[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()\.,;\s@\"]+\.{0,1})+[^<>()\.,;:\s@\"]{2,})$/.test(email)) {
+      return { statusCode: 400, data: { status: 'error', message: 'Please provide a valid email' } };
```

Comment:
The error formats is a bit unusual. Why is the error message in the `data` property? The sample error format on the readme.md of this project is
```
{
  "errors":{
    "body": [
      "can't be empty"
    ]
  }
}